src/lib/analytics/segmentOneTrustIntegration.ts

```diff
@@ -0,0 +1,179 @@
+import { data as sd } from "sharify"
+
+const SEGMENT_MAPPING = "C0001" // Segment itself assigned OneTrust Strictly Necessary category.
+const SEGMENT_WRITE_KEY = sd.SEGMENT_WRITE_KEY
+
+let ONETRUST_PREVIOUS_CONSENT = ""
+let SEGMENT_DESTINATIONS = []
+
+export function fetchSegmentDestinationsAndLoad() {
+  console.log("fetch segment destinations")
+  fetchDestinations(SEGMENT_WRITE_KEY).then(destinations => {
+    SEGMENT_DESTINATIONS = destinations
+
+    waitForOneTrust()
+
+    if (oneTrustReady()) {
+      getConsentAndLoadSegment()
```

Comment:
In the event that OneTrust never becomes ready, Segment is not loaded. That seems bad. I will revise this so that in that case we load Segment for `Strictly Necessary` (`C0001`) destinations.

Reply:
revised.